docs/DesignConsiderations.md

```diff
@@ -0,0 +1,84 @@
+# HLSL Design Considerations
+
+When designing and proposing a feature for HLSL there are some design
+considerations that should be taken into account.
+
+## Style Conventions
+
+HLSL's built-in types and methods should conform to a consistent coding style.
+
+* Data types, methods and built-in functions should all be `CamelCase`.
+* Namespaces and keywords are lowercase, `_` separated.
+* Vulkan-specific functionality should be added to the `vk` namespace.
+* Microsoft-style attributes interchangably use `CamelCase` and `_` separation,
+  but should prefer `CamelCase`
+* System Value semantics are case insensitive, should be specified `CamelCase`
+  and prefixed `SV_`.
+
+## Versioning
+
+All features should consider how users can adapt to the presence or absence of
+support. HLSL has two primary revisioning axis: language version and runtime
+version. How the two versions interact is not always a clean and easy line to
```

Comment:
I can't decide if I mind that this mentions the guessable, but not immediately obvious connection between "runtime version" and "shader models". It's explained below and I see why a generic term was used, but perhaps something like "(e.g. shader models) might make it clearer sooner.

Reply:
I specifically chose not to use the term "shader models" because it is DX-specific, and doesn't capture the Vulkan version angle.